### ISSUE TITLE
The query builder seems to be dropping field type information from aggregate columns in nested questions if you use the summarize sidebar

### ISSUE THREAD

**Issue by kdoh** [metabase/metabase#15397]
**Describe the bug**
When trying to use an aggregate field in a nested query off of a native question we seem to be dropping `field/type` information about that field, making it impossible to aggregate or break out the question further using the Summarize sidebar.

**To Reproduce**

https://user-images.githubusercontent.com/5248953/113012900-32737680-9149-11eb-91fd-5ac2cf31cabd.mp4

1. Create a saved SQL question with the following SQL
```sql
select count(*),  orders.product_id from orders group by orders.product_id; 
```
2. Start a new question by going to "Ask a question" -> "Saved questions" and select the question you made in step 1
3. Try using the "Count (*)" column in an aggregation OR breakout using the Summarize sidebar
4. We get an error back like this

Aggregation use:
```
Value does not match schema: {:query {:aggregation [(named [nil (named (named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field") "field-or-expression")] "Must be a valid instance of one of these clauses: :count, :avg, :cum-count, :cum-sum, :distinct, :stddev, :sum, :min, :max, :metric, :share, :count-where, :sum-where, :case, :median, :percentile, :var")], :breakout [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}
```
Breakout use:
```
Value does not match schema: {:query {:breakout [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}
```

5. Interestingly, if you use the header actions on the "Count (*)" column (trying to take the average for example), everything works as expected, indicating to me that we're doing something differently with fields in the sidebar.

**Logs**
The request made to the server looks like this: 

Trying an aggregation: 
```json
{
  "database":1,
  "query":{
     "source-table":"card__2222",
     "aggregation":[
         ["avg",
         ["field","COUNT(*)",null]]]
      },
     "type":"query",
     "parameters":[]
}
```

Trying a breakout:
```json
{
    "type": "query",
    "query": {
        "source-table": "card__2222",
        "aggregation": [
            [
                "count"
            ]
        ],
        "breakout": [
            [
                "field",
                "COUNT(*)",
                {
                    "binning": {
                        "strategy": "default"
                    }
                }
            ]
        ]
    },
    "database": 1,
    "parameters": []
}
```

The working request from the column header actions looks like this - note the addition of the "type" information

```json
{"type":"query","query":{"source-table":"card__2222","aggregation":[["avg",["field","COUNT(*)",{"base-type":"type/BigInteger"}]]]},"database":1,"parameters":[]}
```

**Expected behavior**
I'd expect to be able to use the basic aggregation functions on top of my saved native question.

**Information about your Metabase Installation:**
```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:88.0) Gecko/20100101 Firefox/88.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "4.14.219-119.340.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "postgres",
      "bigquery",
      "redshift",
      "druid",
      "mongo",
      "googleanalytics",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "9.6.20"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-03-29",
      "tag": "v0.39.0-SNAPSHOT",
      "branch": "master",
      "hash": "795c172"
    },
    "settings": {
      "report-timezone": "US/Pacific"
    }
  }
}
```

**Severity**
Pretty severe - this is [preventing us from shipping a new feature for 0.39](https://github.com/metabase/metabase/pull/15377) and degrades the experience of nested questions generally.

**Comment by flamber** [comment]
Same problem as #15725, which was fixed by #16140 and will be part of 0.39.3

**Comment by nemanjaglumac** [comment]
Repro for this issue is in `master` and cannot be unskipped currently. We'll have to wait until the next merge.
